src/shared/awsagent/roles.go

```diff
@@ -98,23 +103,42 @@ func (a *Agent) DeleteOtterizeIAMRole(ctx context.Context, accountName string) e
 	}
 
 	_, found := lo.Find(role.Tags, func(tag types.Tag) bool {
-		return *tag.Key == serviceAccountNameTagKey
+		if *tag.Key == serviceAccountNameTagKey && *tag.Value == accountName {
+			return true
+		}
+		if *tag.Key == serviceAccountNamespaceTagKey && *tag.Value == namespaceName {
+			return true
+		}
+
+		return false
 	})
 
 	if !found {
-		errorMessage := fmt.Sprintf("refusing to delete role that's not tagged with %s", serviceAccountNameTagKey)
-		logger.Error(errorMessage)
+		errorMessage := "refusing to delete role that's not tagged with with appropriate tags"
+		logger.WithField("arn", role.Arn).Error(errorMessage)
 		return errors.New(errorMessage)
 	}
 
-	_, err = a.iamClient.DeleteRolePolicy(ctx, &iam.DeleteRolePolicyInput{
+	_, err = a.iamClient.GetRolePolicy(ctx, &iam.GetRolePolicyInput{
 		PolicyName: role.RoleName,
 		RoleName:   role.RoleName,
 	})
 
-	if err != nil {
-		logger.WithError(err).Errorf("failed to delete inline policy")
-		return err
+	if err == nil {
```

Comment:
Reverse the condition - early return using err != nil instead of nested conditions